avocado/core/mux.py

```diff
@@ -163,15 +164,25 @@ def initialize_mux(self, root, mux_path, debug):
         self.root = root
         self.mux_path = mux_path
         self.debug = debug
+        self.variant_ids = self._get_variant_ids()
+
+    def _get_variant_ids(self):
+        variant_ids = []
+        for variant in MuxTree(self.root):
+            variant.sort(key=lambda x: x.path)
+            fingerprint = "-".join(_.fingerprint() for _ in variant)
+            variant_ids.append("-".join(node.name for node in variant) + '-' +
```

Comment:
democracy is also not perfect, but so far it seems to be the best system... I played with it, tried some (avocado-vt) based yaml files and it looked significantly better than indexes. I tried using full paths, but that produced too long names.

For the future I see a possibility of another tag to say "this should be included in variant name" vs. "this leaf should not be included in variant name" which might improve the readability (inspired by short_name from avocado-vt), or even a way to override the variant name from the content (eg. look for certain value in each leaf, if exists use it instead of the leaf name...).

Anyway for now I think this is a good start.